File: 55.py
def back(x):
    sum = 0
    for i in range(0,len(x)):
        sum = sum + x[i]*10**(len(x)-i-1)
    return(sum)

def linn(x, count):
    num = [int(d) for d in str(x)]
    numrev = num[::-1]
    numrev1 = back(numrev)
    new = numrev1 + x
    num2 = [int(d) for d in str(new)]
    num2rev = num2[::-1]
    if num2 == num2rev:
        return(1)
    if count == 50:
        return(0)
    else:
        return(linn(new, count + 1))

total = 0
for k in range(1,10001):
    total = total + linn(k, 0)

File: 55.R
library(TeachingDemos)

back = function(x){
  n = length(x)
  sum = 0
  for (m in 1:n){
    sum = sum + x[m]*10^(n-m)
  }
  return(sum)
}

linn = function(x, count){
  dig = digits(x)
  digrev = rev(dig)
  new = x + back(digrev)
  newdig = digits(new)
  newdigrev = rev(newdig)
  if (all.equal(newdig, newdigrev)==TRUE){
    return(1)
  }
  if (count == 50){
    return(0)
  }
  else{
    return(linn(new,count+1))
  }
}

s = 0 

for (k in 1:10000){
  s = s + linn(k, 1)
}